File: backend/app/schemas/treatment.py
from app.schemas.patient_owner import PatientResponse
from app.schemas.template_measure import TemplateResponse
from pydantic import BaseModel
from uuid import UUID
from datetime import datetime
from typing import Optional, List

# --- Log Value Schemas ---


class LogValueBase(BaseModel):
    measure_id: UUID
    value: Optional[str] = None


class LogValueCreate(LogValueBase):
    pass


class LogValueResponse(LogValueBase):
    id: UUID
    treatment_log_id: UUID

    class Config:
        from_attributes = True

# --- Treatment Log Schemas ---


class TreatmentLogBase(BaseModel):
    notes: Optional[str] = None


class TreatmentLogCreate(TreatmentLogBase):
    values: Optional[List[LogValueCreate]] = []
    logged_at: Optional[datetime] = None


class TreatmentLogResponse(TreatmentLogBase):
    id: UUID
    treatment_session_id: UUID
    logged_at: datetime
    values: Optional[List[LogValueResponse]] = []

    class Config:
        from_attributes = True

# --- Treatment Session Schemas ---


class TreatmentSessionBase(BaseModel):
    patient_id: UUID
    template_id: UUID
    status: Optional[str] = "active"
    notes: Optional[str] = None


class TreatmentSessionCreate(TreatmentSessionBase):
    pass


class TreatmentSessionUpdate(BaseModel):
    status: Optional[str] = None
    notes: Optional[str] = None
    completed_at: Optional[datetime] = None


class TreatmentSessionResponse(TreatmentSessionBase):
    id: UUID
    started_at: datetime
    completed_at: Optional[datetime] = None
    logs: Optional[List[TreatmentLogResponse]] = []
    patient: Optional[PatientResponse] = None
    template: Optional[TemplateResponse] = None

    class Config:
        from_attributes = True

File: frontend/src/types/treatment.ts
import type { Patient } from "./patient";
import type { Template } from "./template";

export interface LogValueBase {
  measure_id: string;
  value?: string | null;
}

export type LogValueCreate = LogValueBase;

export interface LogValue extends LogValueBase {
  id: string;
  treatment_log_id: string;
}

export interface TreatmentLogBase {
  notes?: string | null;
}

export interface TreatmentLogCreate extends TreatmentLogBase {
  values?: LogValueCreate[];
  logged_at?: string | null;
}

export interface TreatmentLog extends TreatmentLogBase {
  id: string;
  treatment_session_id: string;
  logged_at: string;
  values?: LogValue[];
}

export interface TreatmentSessionBase {
  patient_id: string;
  template_id: string;
  status?: string | null; // e.g., 'active'
  notes?: string | null;
}

export type TreatmentSessionCreate = TreatmentSessionBase;

export interface TreatmentSession extends TreatmentSessionBase {
  id: string;
  started_at: string;
  completed_at?: string | null;
  logs?: TreatmentLog[];
  patient?: Patient | null;
  template?: Template | null;
}
